lib/game.rb

```diff
@@ -0,0 +1,85 @@
+class Game
+  attr_accessor :board, :player_1, :player_2, :name
+
+  # Win combinations constant
+  WIN_COMBINATIONS = [
+    [0, 1, 2], # Top row
+    [3, 4, 5], # Middle row
+    [6, 7, 8], # Bottom row
+    [0, 3, 6], # Left column
+    [1, 4, 7], # Middle column
+    [2, 5, 8], # Right column
+    [0, 4, 8], # Left Diagonal
+    [2, 4, 6]  # Right Diagonal
+  ]
+  
+  # Game methods
+  def initialize(player_1 = Players::Human.new("X", name), player_2 = Players::Human.new("O", name), board = Board.new)
+    @board = board
+    @player_1 = player_1
+    @player_2 = player_2
+    self
+  end
+
+  def current_player
+    board.turn_count.even? ? player_1 : player_2
+  end
+
+  def won?
+    # THIS FUCKING SUCKEDDDD
+    win = false
+    WIN_COMBINATIONS.each do |combo|
+      win = combo if check_combo(combo)
+    end
+    win
+  end
+
+  # #won? helper method
+  def check_combo(combo)
+    win = false
+    poss_win = []
+    combo.each {|i| poss_win << board.cells[i]}
+    win = combo if poss_win == ["X", "X", "X"] || poss_win == ["O", "O", "O"]
+    win
+  end
+
+  def draw?
+    board.full? && !won?
+  end
+
+  def over?
+    draw? || won?
+  end
+
+  def winner
+    board.cells[won?&.first] if won?
+  end
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.